elements/lisk-transactions/src/sign.ts

```diff
@@ -0,0 +1,182 @@
+import { BaseTransaction } from './base_transaction';
+import { sortKeysAscending } from './utils';
+import { validateTransactionSchema } from './validate';
+
+interface MultiSignatureKeys {
+	readonly mandatoryKeys: Array<Readonly<Buffer>>;
+	readonly optionalKeys: Array<Readonly<Buffer>>;
+}
+
+export const getSigningBytes = (
+	assetSchema: object,
+	transactionObject: Record<string, unknown>,
+): Buffer => {
+	if (typeof transactionObject.asset !== 'object' || transactionObject.asset === null) {
+		throw new Error('Asset must be of type object and not null');
+	}
+	validateTransactionSchema(assetSchema, transactionObject);
+	const assetBytes = codec.encode((assetSchema as unknown) as Schema, transactionObject.asset);
+	const transactionBytes = codec.encode(BaseTransaction.BASE_SCHEMA, {
+		...transactionObject,
+		asset: assetBytes,
+		signatures: [],
+	});
+
+	return transactionBytes;
+};
+
+export const signTransaction = (
+	assetSchema: object,
+	transactionObject: Record<string, unknown>,
+	networkIdentifier: Buffer,
+	senderPassphrase: string,
+): Record<string, unknown> => {
+	if (!networkIdentifier.length) {
+		throw new Error('Network identifier is required to sign a transaction');
+	}
+
+	if (!senderPassphrase) {
+		throw new Error('Passphrase is required to sign a transaction');
+	}
+	validateTransactionSchema(assetSchema, transactionObject);
+	const { publicKey } = getAddressAndPublicKeyFromPassphrase(senderPassphrase);
+
+	if (
+		!Buffer.isBuffer(transactionObject.senderPublicKey) ||
+		!transactionObject.senderPublicKey.equals(publicKey)
+	) {
+		throw new Error('Transaction senderPublicKey does not match public key from passphrase');
+	}
+
+	const transactionWithNetworkIdentifierBytes = Buffer.concat([
+		networkIdentifier,
+		getSigningBytes(assetSchema, transactionObject),
+	]);
+
+	const signature = signData(transactionWithNetworkIdentifierBytes, senderPassphrase);
+	// Reset signatures
+	if (!Array.isArray(transactionObject.signatures)) {
+		throw new Error('Signatures must be of type array');
+	}
+	transactionObject.signatures.push(signature);
+	return transactionObject;
+};
+
+const sanitizeSignaturesArray = (
+	transactionObject: Record<string, unknown>,
+	keys: MultiSignatureKeys,
+	includeSenderSignature: boolean,
+): void => {
+	let numberOfSignatures = keys.mandatoryKeys.length + keys.optionalKeys.length;
+	// Add one extra for multisig account registration
+	if (includeSenderSignature) {
+		numberOfSignatures += 1;
+	}
+
+	for (let i = 0; i < numberOfSignatures; i += 1) {
+		if (
+			Array.isArray(transactionObject.signatures) &&
+			transactionObject.signatures[i] === undefined
+		) {
+			// eslint-disable-next-line no-param-reassign
+			transactionObject.signatures[i] = Buffer.from('');
+		}
+	}
+};
+
+export const signMultiSignatureTransaction = (
+	assetSchema: object,
+	transactionObject: Record<string, unknown>,
+	networkIdentifier: Buffer,
+	senderPassphrase: string,
```

Comment:
```suggestion
	passphrase: string,
```
since it might not be sender